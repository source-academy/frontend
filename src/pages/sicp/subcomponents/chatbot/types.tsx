type ChatMessage = {
  role: 'user' | 'assistant';
  content: string;
};

type InitChatResponse = {
  response: ChatMessage;
  conversationId: string;
};

type ContinueChatResponse = {
  response: string;
  conversationId: string;
};
