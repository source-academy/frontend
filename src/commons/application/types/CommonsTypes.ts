import { Router } from '@remix-run/router';

export const LOG_OUT = 'LOG_OUT';
export const UPDATE_REACT_ROUTER = 'UPDATE_REACT_ROUTER';

export type RouterState = Router | null;
