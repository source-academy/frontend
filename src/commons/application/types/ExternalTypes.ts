/**
 * Defined for displaying an external library.
 * @see Library the definition of a Library at AssessmentTypes.ts in an assessment.
 */
export type External = {
  key: number;
  name: ExternalLibraryName;
  symbols: string[];
};

export enum ExternalLibraryName {
  NONE = 'NONE',
  SOUNDS = 'SOUNDS',
  EV3 = 'EV3' // Remote execution device library
}

export type ExternalLibrary = {
  name: ExternalLibraryName;
  symbols: string[];
};

/**
 * Defines which external libraries are available for usage, and what
 * external symbols (exposed functions) are under them.
 */

const libEntries: Array<[ExternalLibraryName, string[]]> = [
  [ExternalLibraryName.NONE, []],
  [ExternalLibraryName.SOUNDS, []]
];

export const externalLibraries: Map<string, string[]> = new Map(libEntries);
