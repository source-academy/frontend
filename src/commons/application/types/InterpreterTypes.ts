export const BEGIN_INTERRUPT_EXECUTION = 'BEGIN_INTERRUPT_EXECUTION';
export const END_INTERRUPT_EXECUTION = 'END_INTERRUPT_EXECUTION';
export const EVAL_INTERPRETER_ERROR = 'EVAL_INTERPRETER_ERROR';
export const EVAL_INTERPRETER_SUCCESS = 'EVAL_INTERPRETER_SUCCESS';
export const EVAL_TESTCASE_FAILURE = 'EVAL_TESTCASE_FAILURE';
export const EVAL_TESTCASE_SUCCESS = 'EVAL_TESTCASE_SUCCESS';
export const HANDLE_CONSOLE_LOG = 'HANDLE_CONSOLE_LOG';
export const BEGIN_DEBUG_PAUSE = 'BEGIN_DEBUG_PAUSE';
export const END_DEBUG_PAUSE = 'END_DEBUG_PAUSE';
export const DEBUG_RESUME = 'DEBUG_RESUME';
export const DEBUG_RESET = 'DEBUG_RESET';
export const UPDATE_EDITOR_HIGHLIGHTED_LINES = 'UPDATE_EDITOR_HIGHLIGHTED_LINES';
