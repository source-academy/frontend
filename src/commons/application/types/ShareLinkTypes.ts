export type ShareLinkShortenedUrlResponse = {
  shortenedUrl: string;
};
