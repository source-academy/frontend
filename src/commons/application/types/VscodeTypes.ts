export type VscodeState = {
  isVscode: boolean;
};
