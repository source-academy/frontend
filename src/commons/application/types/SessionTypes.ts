import { Octokit } from '@octokit/rest';
import { Chapter, Variant } from 'js-slang/dist/types';

import { Grading, GradingOverview } from '../../../features/grading/GradingTypes';
import { Device, DeviceSession } from '../../../features/remoteExecution/RemoteExecutionTypes';
import {
  Assessment,
  AssessmentConfiguration,
  AssessmentOverview
} from '../../assessment/AssessmentTypes';
import { MissionRepoData } from '../../githubAssessments/GitHubMissionTypes';
import { Notification } from '../../notificationBadge/NotificationBadgeTypes';
import { GameState, Role, Story } from '../ApplicationTypes';

export const FETCH_AUTH = 'FETCH_AUTH';
export const FETCH_USER_AND_COURSE = 'FETCH_USER_AND_COURSE';
export const FETCH_COURSE_CONFIG = 'FETCH_COURSE_CONFIG';
export const FETCH_ASSESSMENT = 'FETCH_ASSESSMENT';
export const FETCH_ASSESSMENT_ADMIN = 'FETCH_ASSESSMENT_ADMIN';
export const FETCH_ASSESSMENT_OVERVIEWS = 'FETCH_ASSESSMENT_OVERVIEWS';
export const FETCH_ALL_USER_XP = 'FETCH_ALL_USER_XP';
export const FETCH_TOTAL_XP = 'FETCH_TOTAL_XP';
export const FETCH_TOTAL_XP_ADMIN = 'FETCH_TOTAL_XP_ADMIN';
export const FETCH_GRADING = 'FETCH_GRADING';
export const FETCH_GRADING_OVERVIEWS = 'FETCH_GRADING_OVERVIEWS';
export const LOGIN = 'LOGIN';
export const LOGOUT_GOOGLE = 'LOGOUT_GOOGLE';
export const LOGIN_GITHUB = 'LOGIN_GITHUB';
export const LOGOUT_GITHUB = 'LOGOUT_GITHUB';
export const SET_TOKENS = 'SET_TOKENS';
export const SET_USER = 'SET_USER';
export const SET_COURSE_CONFIGURATION = 'SET_COURSE_CONFIGURATION';
export const SET_COURSE_REGISTRATION = 'SET_COURSE_REGISTRATION';
export const SET_ASSESSMENT_CONFIGURATIONS = 'SET_ASSESSMENT_CONFIGURATIONS';
export const SET_ADMIN_PANEL_COURSE_REGISTRATIONS = 'SET_ADMIN_PANEL_COURSE_REGISTRATIONS';
export const SET_GOOGLE_USER = 'SET_GOOGLE_USER';
export const SET_GITHUB_ASSESSMENT = 'SET_GITHUB_ASSESSMENT';
export const SET_GITHUB_OCTOKIT_OBJECT = 'SET_GITHUB_OCTOKIT_OBJECT';
export const SET_GITHUB_ACCESS_TOKEN = 'SET_GITHUB_ACCESS_TOKEN';
export const SUBMIT_ANSWER = 'SUBMIT_ANSWER';
export const SUBMIT_ASSESSMENT = 'SUBMIT_ASSESSMENT';
export const SUBMIT_GRADING = 'SUBMIT_GRADING';
export const SUBMIT_GRADING_AND_CONTINUE = 'SUBMIT_GRADING_AND_CONTINUE';
export const REAUTOGRADE_SUBMISSION = 'REAUTOGRADE_SUBMISSION';
export const REAUTOGRADE_ANSWER = 'REAUTOGRADE_ANSWER';
export const REMOVE_GITHUB_OCTOKIT_OBJECT_AND_ACCESS_TOKEN =
  'REMOVE_GITHUB_OCTOKIT_OBJECT_AND_ACCESS_TOKEN';
export const UNSUBMIT_SUBMISSION = 'UNSUBMIT_SUBMISSION';
export const UPDATE_ASSESSMENT_OVERVIEWS = 'UPDATE_ASSESSMENT_OVERVIEWS';
export const UPDATE_TOTAL_XP = 'UPDATE_TOTAL_XP';
export const UPDATE_ALL_USER_XP = 'UPDATE_ALL_USER_XP';
export const UPDATE_ASSESSMENT = 'UPDATE_ASSESSMENT';
export const UPDATE_GRADING_OVERVIEWS = 'UPDATE_GRADING_OVERVIEWS';
export const UPDATE_GRADING = 'UPDATE_GRADING';
export const FETCH_NOTIFICATIONS = 'FETCH_NOTIFICATIONS';
export const ACKNOWLEDGE_NOTIFICATIONS = 'ACKNOWLEDGE_NOTIFICATIONS';
export const UPDATE_NOTIFICATIONS = 'UPDATE_NOTIFICATIONS';
export const UPDATE_LATEST_VIEWED_COURSE = 'UPDATE_LATEST_VIEWED_COURSE';
export const UPDATE_COURSE_CONFIG = 'UPDATE_COURSE_CONFIG';
export const FETCH_ASSESSMENT_CONFIGS = 'FETCH_ASSESSMENT_CONFIGS';
export const UPDATE_ASSESSMENT_CONFIGS = 'UPDATE_ASSESSMENT_CONFIGS';
export const DELETE_ASSESSMENT_CONFIG = 'DELETE_ASSESSMENT_CONFIG';
export const FETCH_ADMIN_PANEL_COURSE_REGISTRATIONS = 'FETCH_ADMIN_PANEL_COURSE_REGISTRATIONS';
export const UPDATE_USER_ROLE = 'UPDATE_USER_ROLE';
export const UPDATE_COURSE_RESEARCH_AGREEMENT = 'UPDATE_COURSE_RESEARCH_AGREEMENT';
export const DELETE_USER_COURSE_REGISTRATION = 'DELETE_USER_COURSE_REGISTRATION';

export const UPLOAD_KEYSTROKE_LOGS = 'UPLOAD_KEYSTROKE_LOGS';
export const UPLOAD_UNSENT_LOGS = 'UPLOAD_UNSENT_LOGS';

export type SessionState = {
  // Tokens
  readonly accessToken?: string;
  readonly refreshToken?: string;

  // User
  readonly userId?: number;
  readonly name?: string;
  readonly courses: UserCourse[];

  // Course Registration
  readonly courseRegId?: number;
  readonly role?: Role;
  readonly group: string | null;
  readonly gameState: GameState;
  readonly courseId?: number;
  readonly xp: number;
  readonly allUserXp: string[][] | undefined;
  readonly story: Story;

  // Course Configuration
  readonly courseName?: string;
  readonly courseShortName?: string;
  readonly viewable?: boolean;
  readonly enableGame?: boolean;
  readonly enableAchievements?: boolean;
  readonly enableSourcecast?: boolean;
  readonly sourceChapter?: number;
  readonly sourceVariant?: Variant;
  readonly moduleHelpText?: string;
  readonly assetsPrefix?: string;

  readonly assessmentConfigurations?: AssessmentConfiguration[];
  readonly userCourseRegistrations?: AdminPanelCourseRegistration[];

  // For research data collection
  readonly agreedToResearch?: boolean | null;
  readonly sessionId: number;

  readonly assessmentOverviews?: AssessmentOverview[];
  readonly assessments: Map<number, Assessment>;
  readonly gradingOverviews?: GradingOverview[];
  readonly gradings: Map<number, Grading>;
  readonly notifications: Notification[];
  readonly googleUser?: string;
  readonly githubAssessment?: MissionRepoData;
  readonly githubOctokitObject: { octokit: Octokit | undefined };
  readonly githubAccessToken?: string;
  readonly remoteExecutionDevices?: Device[];
  readonly remoteExecutionSession?: DeviceSession;
};

export type Tokens = {
  accessToken: string;
  refreshToken: string;
};

export type UserCourse = {
  courseId: number;
  courseName: string;
  courseShortName: string;
  role: Role;
  viewable: boolean;
};

export type User = {
  userId: number;
  name: string;
  courses: UserCourse[];
};

export type CourseRegistration = {
  courseRegId: number;
  role: Role;
  group: string | null;
  gameState?: GameState;
  courseId: number;
  xp: number;
  story?: Story;
  agreedToResearch: boolean | null;
};

export type CourseConfiguration = {
  courseName: string;
  courseShortName: string;
  viewable: boolean;
  enableGame: boolean;
  enableAchievements: boolean;
  enableSourcecast: boolean;
  sourceChapter: Chapter;
  sourceVariant: Variant;
  moduleHelpText: string;
  assetsPrefix: string;
};

export type AdminPanelCourseRegistration = {
  courseRegId: number;
  courseId: number;
  name?: string;
  username: string;
  role: Role;
  group?: string;
};

export type UpdateCourseConfiguration = Partial<CourseConfiguration>;
