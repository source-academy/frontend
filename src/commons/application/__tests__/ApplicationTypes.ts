import { Chapter, Variant } from 'js-slang/dist/types';

import {
  ALL_LANGUAGES,
  getLanguageConfig,
  pyLanguages,
  schemeLanguages,
  sourceLanguages
} from '../ApplicationTypes';

describe('getLanguageConfig', () => {
  test('works for existing variants', () => {
    for (const language of ALL_LANGUAGES) {
      expect(getLanguageConfig(language.chapter, language.variant)).toEqual(language);
    }
  });

  test('throws an error for an invalid chapter/variant combination', () => {
    expect(() => getLanguageConfig(5, Variant.DEFAULT)).toThrowErrorMatchingSnapshot();
  });
});

describe('available Source language configurations', () => {
  test('matches expected configurations', () => {
    const expectedSourceConfigs = [
      // Source 1
      { chapter: Chapter.SOURCE_1, variant: Variant.DEFAULT, supports: { substVisualizer: true } },
      { chapter: Chapter.SOURCE_1, variant: Variant.TYPED },
      { chapter: Chapter.SOURCE_1, variant: Variant.WASM },
      { chapter: Chapter.SOURCE_1, variant: Variant.LAZY },
      { chapter: Chapter.SOURCE_1, variant: Variant.NATIVE, supports: { substVisualizer: true } },
      // Source 2
      {
        chapter: Chapter.SOURCE_2,
        variant: Variant.DEFAULT,
        supports: { dataVisualizer: true, substVisualizer: true }
      },
      { chapter: Chapter.SOURCE_2, variant: Variant.TYPED, supports: { dataVisualizer: true } },
      { chapter: Chapter.SOURCE_2, variant: Variant.LAZY, supports: { dataVisualizer: true } },
      {
        chapter: Chapter.SOURCE_2,
        variant: Variant.NATIVE,
        supports: { dataVisualizer: true, substVisualizer: true }
      },
      // Source 3
      {
        chapter: Chapter.SOURCE_3,
        variant: Variant.DEFAULT,
        supports: { dataVisualizer: true, envVisualizer: true }
      },
      {
        chapter: Chapter.SOURCE_3,
        variant: Variant.TYPED,
        supports: { dataVisualizer: true, envVisualizer: true }
      },
      {
        chapter: Chapter.SOURCE_3,
        variant: Variant.CONCURRENT,
        supports: { dataVisualizer: true }
      },
      { chapter: Chapter.SOURCE_3, variant: Variant.NON_DET, supports: { dataVisualizer: true } },
      {
        chapter: Chapter.SOURCE_3,
        variant: Variant.NATIVE,
        supports: { dataVisualizer: true, envVisualizer: true }
      },
      // Source 4
      {
        chapter: Chapter.SOURCE_4,
        variant: Variant.DEFAULT,
        supports: { dataVisualizer: true, envVisualizer: true }
      },
      {
        chapter: Chapter.SOURCE_4,
        variant: Variant.TYPED,
        supports: { dataVisualizer: true, envVisualizer: true }
      },
      {
        chapter: Chapter.SOURCE_4,
        variant: Variant.GPU,
        supports: { dataVisualizer: true, envVisualizer: true }
      },
      {
        chapter: Chapter.SOURCE_4,
        variant: Variant.NATIVE,
        supports: { dataVisualizer: true, envVisualizer: true }
      },
      {
        chapter: Chapter.SOURCE_4,
        variant: Variant.EXPLICIT_CONTROL,
        supports: { dataVisualizer: true, envVisualizer: true }
      }
    ];

    expect(sourceLanguages).toMatchObject(expectedSourceConfigs);
  });

  test('matches snapshot', () => {
    expect(sourceLanguages).toMatchSnapshot();
  });
});

describe('available Python language configurations', () => {
  test('matches snapshot', () => {
    expect(pyLanguages).toMatchSnapshot();
  });
});

describe('available Scheme language configurations', () => {
  test('matches snapshot', () => {
    expect(schemeLanguages).toMatchSnapshot();
  });
});
