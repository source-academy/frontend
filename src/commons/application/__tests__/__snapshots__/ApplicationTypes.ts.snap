// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`available Python language configurations matches snapshot 1`] = `
Array [
  Object {
    "chapter": -4,
    "displayName": "Python §1",
    "mainLanguage": "Python",
    "supports": Object {
      "repl": true,
    },
    "variant": "default",
  },
]
`;

exports[`available Scheme language configurations matches snapshot 1`] = `
Array [
  Object {
    "chapter": -9,
    "displayName": "Scheme §1",
    "mainLanguage": "Scheme",
    "supports": Object {
      "repl": true,
    },
    "variant": "default",
  },
  Object {
    "chapter": -10,
    "displayName": "Scheme §2",
    "mainLanguage": "Scheme",
    "supports": Object {
      "repl": true,
    },
    "variant": "default",
  },
  Object {
    "chapter": -11,
    "displayName": "Scheme §3",
    "mainLanguage": "Scheme",
    "supports": Object {
      "repl": true,
    },
    "variant": "default",
  },
  Object {
    "chapter": -12,
    "displayName": "Scheme §4",
    "mainLanguage": "Scheme",
    "supports": Object {
      "repl": true,
    },
    "variant": "default",
  },
  Object {
    "chapter": -13,
    "displayName": "Full Scheme",
    "mainLanguage": "Scheme",
    "supports": Object {
      "repl": true,
    },
    "variant": "default",
  },
]
`;

exports[`available Source language configurations matches snapshot 1`] = `
Array [
  Object {
    "chapter": 1,
    "displayName": "Source §1",
    "mainLanguage": "JavaScript",
    "supports": Object {
      "dataVisualizer": false,
      "envVisualizer": false,
      "multiFile": false,
      "repl": true,
      "substVisualizer": true,
    },
    "variant": "default",
  },
  Object {
    "chapter": 1,
    "displayName": "Source §1 Typed",
    "mainLanguage": "JavaScript",
    "supports": Object {
      "dataVisualizer": false,
      "envVisualizer": false,
      "multiFile": false,
      "repl": true,
      "substVisualizer": false,
    },
    "variant": "typed",
  },
  Object {
    "chapter": 1,
    "displayName": "Source §1 WebAssembly",
    "mainLanguage": "JavaScript",
    "supports": Object {
      "dataVisualizer": false,
      "envVisualizer": false,
      "multiFile": false,
      "repl": true,
      "substVisualizer": false,
    },
    "variant": "wasm",
  },
  Object {
    "chapter": 1,
    "displayName": "Source §1 Lazy",
    "mainLanguage": "JavaScript",
    "supports": Object {
      "dataVisualizer": false,
      "envVisualizer": false,
      "multiFile": false,
      "repl": true,
      "substVisualizer": false,
    },
    "variant": "lazy",
  },
  Object {
    "chapter": 1,
    "displayName": "Source §1 Native",
    "mainLanguage": "JavaScript",
    "supports": Object {
      "dataVisualizer": false,
      "envVisualizer": false,
      "multiFile": false,
      "repl": true,
      "substVisualizer": true,
    },
    "variant": "native",
  },
  Object {
    "chapter": 2,
    "displayName": "Source §2",
    "mainLanguage": "JavaScript",
    "supports": Object {
      "dataVisualizer": true,
      "envVisualizer": false,
      "multiFile": true,
      "repl": true,
      "substVisualizer": true,
    },
    "variant": "default",
  },
  Object {
    "chapter": 2,
    "displayName": "Source §2 Typed",
    "mainLanguage": "JavaScript",
    "supports": Object {
      "dataVisualizer": true,
      "envVisualizer": false,
      "multiFile": true,
      "repl": true,
      "substVisualizer": false,
    },
    "variant": "typed",
  },
  Object {
    "chapter": 2,
    "displayName": "Source §2 Lazy",
    "mainLanguage": "JavaScript",
    "supports": Object {
      "dataVisualizer": true,
      "envVisualizer": false,
      "multiFile": true,
      "repl": true,
      "substVisualizer": false,
    },
    "variant": "lazy",
  },
  Object {
    "chapter": 2,
    "displayName": "Source §2 Native",
    "mainLanguage": "JavaScript",
    "supports": Object {
      "dataVisualizer": true,
      "envVisualizer": false,
      "multiFile": true,
      "repl": true,
      "substVisualizer": true,
    },
    "variant": "native",
  },
  Object {
    "chapter": 3,
    "displayName": "Source §3",
    "mainLanguage": "JavaScript",
    "supports": Object {
      "dataVisualizer": true,
      "envVisualizer": true,
      "multiFile": true,
      "repl": true,
      "substVisualizer": false,
    },
    "variant": "default",
  },
  Object {
    "chapter": 3,
    "displayName": "Source §3 Typed",
    "mainLanguage": "JavaScript",
    "supports": Object {
      "dataVisualizer": true,
      "envVisualizer": true,
      "multiFile": true,
      "repl": true,
      "substVisualizer": false,
    },
    "variant": "typed",
  },
  Object {
    "chapter": 3,
    "displayName": "Source §3 Concurrent",
    "mainLanguage": "JavaScript",
    "supports": Object {
      "dataVisualizer": true,
      "envVisualizer": false,
      "multiFile": true,
      "repl": false,
      "substVisualizer": false,
    },
    "variant": "concurrent",
  },
  Object {
    "chapter": 3,
    "displayName": "Source §3 Non-Det",
    "mainLanguage": "JavaScript",
    "supports": Object {
      "dataVisualizer": true,
      "envVisualizer": false,
      "multiFile": true,
      "repl": true,
      "substVisualizer": false,
    },
    "variant": "non-det",
  },
  Object {
    "chapter": 3,
    "displayName": "Source §3 Native",
    "mainLanguage": "JavaScript",
    "supports": Object {
      "dataVisualizer": true,
      "envVisualizer": true,
      "multiFile": true,
      "repl": true,
      "substVisualizer": false,
    },
    "variant": "native",
  },
  Object {
    "chapter": 4,
    "displayName": "Source §4",
    "mainLanguage": "JavaScript",
    "supports": Object {
      "dataVisualizer": true,
      "envVisualizer": true,
      "multiFile": true,
      "repl": true,
      "substVisualizer": false,
    },
    "variant": "default",
  },
  Object {
    "chapter": 4,
    "displayName": "Source §4 Typed",
    "mainLanguage": "JavaScript",
    "supports": Object {
      "dataVisualizer": true,
      "envVisualizer": true,
      "multiFile": true,
      "repl": true,
      "substVisualizer": false,
    },
    "variant": "typed",
  },
  Object {
    "chapter": 4,
    "displayName": "Source §4 GPU",
    "mainLanguage": "JavaScript",
    "supports": Object {
      "dataVisualizer": true,
      "envVisualizer": true,
      "multiFile": true,
      "repl": true,
      "substVisualizer": false,
    },
    "variant": "gpu",
  },
  Object {
    "chapter": 4,
    "displayName": "Source §4 Native",
    "mainLanguage": "JavaScript",
    "supports": Object {
      "dataVisualizer": true,
      "envVisualizer": true,
      "multiFile": true,
      "repl": true,
      "substVisualizer": false,
    },
    "variant": "native",
  },
  Object {
    "chapter": 4,
    "displayName": "Source §4 Explicit-Control",
    "mainLanguage": "JavaScript",
    "supports": Object {
      "dataVisualizer": true,
      "envVisualizer": true,
      "multiFile": true,
      "repl": true,
      "substVisualizer": false,
    },
    "variant": "explicit-control",
  },
]
`;

exports[`getLanguageConfig throws an error for an invalid chapter/variant combination 1`] = `"Language config not found for chapter 5 variant default"`;
